is_core/contrib/background_export/views.py

```diff
@@ -21,7 +21,7 @@ def get_redirect_url(self, *args, **kwargs):
 
         if (core.permission.has_permission('read_all', self.request, self, obj=exported_file)
                 or (core.permission.has_permission('read_own', self.request, self, obj=exported_file)
-                    and exported_file.created_by == self.request.user)):
```

Comment:
it is weird from two reasons SimpleLazyObject is proxy and __eq__ method is proxied too. Secondly object shoud be evaluated in is_authenticated method which should be called in read_all permission. Are you sure with that? I do not like natural_key method because not every user must be created from BaseUser

Reply:
I am planning to have a second look - my explanation is probably not correct and there may be more solutions then using `natural_key`, but I am quite confident that at least part of the observed problem lies in this condition. Maybe it is because the user objects being compared are not necessarily of exactly the same type? (if I remember correctly, `exported_file.created_by` would be instance of AUTH_USER_MODEL (settings.dependent, user.User in our case) and self.request.user would be BackendUser?)

Reply:
At this moment I see it as kind of a weird problem - this part of is-core is generic, but should be able to compare/handle any user models we don't know much about. I also considered getting USERNAME_FIELD for both of the models and comparing those + checking if the users are instances of the same class.

Reply:
Yes, `exported_file.created_by` is (in our case) an instance of <class 'common.apps.users.models.User'>, while self.request.user is <class 'common.apps.users.models.BackendUser'>.

Reply:
+ we use the multi-domain (ADMIN_SITE etc.) setting and user models are defined in our own models, on a per-domain basis, making this check particularly difficult if the solution should be a general one.

Reply:
Another approach (that also works) is to leave this condition as is (no changes to is-core) and writing `__eq__` method for the base User model that will basically handle the comparison instead of this view. That also means each user of is-core will have to write their own comparison method if their request user is different from AUTH_USER_MODEL.

Reply:
this is complicated. Use check by pk please. In the django config is used default User model and we must trust user backend to return only user model. Therefore in the request.user can be (AnonymousUser with None pk, User model or its descendants.